lib/vsc/install/shared_setup.py

```diff
@@ -1254,13 +1278,15 @@ def parse_target(self, target, urltemplate=None):
 
         # update the cmdclass with ones from vsc_setup_klass
         # cannot do this in one go, when SHARED_TARGET is defined, vsc_setup doesn't exist yet
+        keepers = new_target.copy()
         for name, klass in new_target['cmdclass'].items():
             try:
                 new_target['cmdclass'][name] = getattr(vsc_setup_klass, klass.__name__)
             except AttributeError:
-                del new_target['cmdclass'][name]
+                print("Not including new_target['cmdclass']['%s']" %name)
 
         # prepare classifiers
+        new_target = keepers
```

Comment:
Wait, this isn't correct I think, stuff is missing above to ensure the semantics haven't changed?

Reply:
yes we should be updating "keepers," which is an initial copy of new_target.  Before the strategy was to remove things from new_target as we were looping, which resulted in weirdness.